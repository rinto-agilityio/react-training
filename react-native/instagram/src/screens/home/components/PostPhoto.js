// Libs
import React from 'react'
import { Image } from 'react-native'
import PropTypes from 'prop-types'

// Styles
import { styles } from './styles/PostPhotoStyles'

const PostPhoto = ({ uri }) => (
  <Image
    resizeMode={'cover'}
    style={styles.photo}
    source={{ uri: uri }}
  />
)

PostPhoto.defaultProps = {
  uri: ''
}

PostPhoto.propTypes = {
  uri: PropTypes.string
}

export default PostPhoto
