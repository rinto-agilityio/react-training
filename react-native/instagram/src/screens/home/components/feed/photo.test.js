// Components
import PostPhoto from './Photo'

// Mocking data
import { photos } from '@test/__mocks__/sample-data'

describe('PostPhoto component', () => {
  const singlePost = photos[1]

  it('Renders correctly', () => {
    const treeDOM = renderer
      .create(<PostPhoto display_url={singlePost.display_url} />)
      .toJSON()

    expect(treeDOM).toMatchSnapshot()
  })
})
