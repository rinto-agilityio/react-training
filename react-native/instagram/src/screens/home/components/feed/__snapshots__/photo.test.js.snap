// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostPhoto component Renders correctly 1`] = `
<Image
  resizeMode="cover"
  source={
    Object {
      "uri": "",
    }
  }
  style={
    Object {
      "height": 300,
      "width": "100%",
    }
  }
/>
`;
